### ISSUE TITLE
Don't fetch all users list when opening the contribute panel

### ISSUE THREAD

**Issue by pgiraud** [hotosm/osm-tasking-manager2#923]
The complete list of users is loaded via the `users.json` request when starting to contribute on a task. This is useful to be able to mention a user in the comment field. This request is heavy and not always required.
See https://github.com/hotosm/osm-tasking-manager2/issues/918#issuecomment-269898668

We should find a better way to handle this and potentially do the filter queries on the server on-the-fly.

**Comment by CloCkWeRX** [comment]
Shifted from comment on #926  - I think that'll help, but we might want further work on the feature - this data supports ```@notifications```; which are very often common to the task or even project you are looking at.

I think the call to ```/stats``` for a project already contains a list of users; potentially loading that to populate the typeahead might be even more efficient

**Comment by pgiraud** [comment]
@CloCkWeRX I agree with you. The list should by default show the users who contributed to the task and then project first.

**Comment by pgiraud** [comment]
Instead of query the whole list of users at once, we should do some server side filtering. `At.js` support server side filtering.
See https://github.com/hotosm/osm-tasking-manager2/pull/926#issuecomment-270066060
